webrtc/README.md

```diff
@@ -161,6 +165,50 @@ MUST pass an empty string. When receiving an `RTCDataChannel` via
 an empty string. This allows future versions of this specification to make use
 of the `RTCDataChannel` `label` property.
 
+## Closing an `RTCDataChannel`
+
+Some WebRTC implementations do not guarantee that any queued messages will be
+sent after a datachannel is closed.  Other implementatations maintain separate
+outgoing message and transport queues, the status of which may not be visible
+to the user. Consequently we must add an additional layer of signaling to ensure
+reliable data delivery.
+
+When a node wishes to close a stream for writing, it SHOULD send a message with
+the `FIN` flag set, then proceed to wait for a `FIN_ACK` message from the remote
+node.
+
+If a `FIN` flag is received and the node has finished writing data, it SHOULD
+respond with a `FIN_ACK` immediately.
+
+If a `FIN` flag is received but the node has not finished writing data, it SHOULD
+delay sending a `FIN_ACK` response until it has finished writing data, sent a
+`FIN` flag, and finally received a `FIN_ACK` in response.
```

Comment:
Why do we need this `SHOULD`? 
The simpler alternative is: 

If a `FIN` flag is received the node should reply with a `FIN_ACK`. If the node has received a `FIN` from the peer and a `FIN_ACK` for the `FIN` it had sent, it SHOULD close the stream. 

```
A -> B: FIN
B -> A: FIN_ACK
B -> A: Data
B -> A: FIN
A -> B: FIN_ACK
```
A closes the stream. if B doesn't receive a FIN_ACK because A is on chromium it's okay, since it has all the information to infer that both sides have closed the stream. It has sent and received a FIN. 

Simultaneous Close case: 

```
A -> B: FIN
B -> A: FIN
A -> B: FIN_ACK
B -> A: FIN_ACK
```
In this case it doesn't matter which side closes the stream on FIN_ACK. if A closes after receiving FIN_ACK from B, at that point B has acked A's FIN and A has received B's FIN. 

cc: @thomaseizinger

Reply:
> Why do we need this `SHOULD`?

I agree that we should not delay the `FIN_ACK`. In my eyes, `FIN_ACK` says nothing about whether you are closing the stream or not, it just acknowledges that you've received the other node's `FIN` which gives the other node that guarantee that they've flushed all their data out of the buffer.

Reply:
This seems better, no need to delay sending the `FIN_ACK` and it seems to work locally.  I've updated the PR text to remove mentions of delaying sending.